src/main/java/org/betonquest/betonquest/compatibility/traincarts/TrainCartsUtils.java

```diff
@@ -0,0 +1,63 @@
+package org.betonquest.betonquest.compatibility.traincarts;
+
+import com.bergerkiller.bukkit.tc.controller.MinecartMember;
+import com.bergerkiller.bukkit.tc.controller.MinecartMemberStore;
+import org.betonquest.betonquest.api.profiles.OnlineProfile;
+import org.bukkit.entity.Entity;
+
+import javax.annotation.Nullable;
+
+/**
+ * Utility class for TrainCarts.
+ */
+public final class TrainCartsUtils {
+
+    /**
+     * This class cannot be instantiated.
+     */
+    private TrainCartsUtils() {
+        // Empty
+    }
+
+    /**
+     * Checks if a profile is currently riding a train from TrainCarts with the specified name.
+     *
+     * @param profile   the {@link OnlineProfile} to check.
+     * @param trainName the name of the train.
+     * @return {@code true} if the player is riding the train with the specified name, otherwise {@code false}.
+     */
+    public static boolean ridesMatchingTrainName(final OnlineProfile profile, final String trainName) {
```

Comment:
I think also this method can be renamed to something like `isRidingTrainCart` because method overloading should make this cleaner